sdk/core/azure-core/src/test/java/com/azure/android/core/common/TestUtils.java

```diff
@@ -1,29 +1,176 @@
+            new ServiceClient.Builder(new OkHttpClient.Builder()
+                .proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress("localhost", 8888)))) :
+            new ServiceClient.Builder();
+
+        for (Interceptor interceptor : interceptors) {
+            serviceBuilder.addInterceptor(interceptor);
+        }
+
+       return serviceBuilder;
+    }
+
+    // ---------------------- CONFIG OPTIONS -----------------------------
+
+    public static boolean useHttps() {
+        return false;
+    }
+
+    public static boolean enableFiddler() {
+        return true;
+    }
+
+    // --------------------- GENERATING TEST RESOURCES ------------------------
+
+    public static String generateResourceName() {
+        return UUID.randomUUID().toString();
+    }
+
+    public static String generateBlockID() {
+        return Base64.getEncoder().encodeToString(UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8));
+    }
+
     public static OkHttpClient buildOkHttpClientWithInterceptor(Interceptor interceptor) {
         return new OkHttpClient().newBuilder()
             .addInterceptor(interceptor)
             .build();
     }
 
+    // -------------------------- HELPER METHODS -----------------------
+
+    public static void awaitOnLatch(CountDownLatch latch, String method) {
+        try {
+            latch.await(10, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            assertFalse(method + " didn't produce any result.", true);
+        }
+    }
+
+    public static boolean validateBasicHeaders(Headers headers) {
+        return headers.get("etag") != null &&
+            // Quotes should be scrubbed from etag header values
+            !headers.get("etag").contains("\"") &&
+            headers.get("last-modified") != null &&
+            headers.get("x-ms-request-id") != null &&
+            headers.get("x-ms-version") != null &&
+            headers.get("date") != null;
+    }
+
+    public static boolean validateBlobProperties(BlobGetPropertiesHeaders headers, String cacheControl, String contentDisposition, String contentEncoding, String contentLanguage, byte[] contentMd5, String contentType) {
+        return Objects.equals(headers.getCacheControl(), cacheControl) &&
+            Objects.equals(headers.getContentDisposition(), contentDisposition) &&
+            Objects.equals(headers.getContentEncoding(), contentEncoding) &&
+            Objects.equals(headers.getContentLanguage(), contentLanguage) &&
+            Arrays.equals(headers.getContentMD5(), contentMd5) &&
+            Objects.equals(headers.getContentType(), contentType);
+    }
+
+    public static String setupMatchCondition(StorageBlobClient client, String containerName, String blobName, String match) {
+        if (receivedEtag.equals(match)) {
+            return client.getBlobProperties(containerName, blobName).getETag();
+        } else {
+            return match;
+        }
+    }
```

Comment:
Should we move any utils methods that are specific to storage or uses storage types to storage-package? core util should not have any dependency on storage or any other azure-sdk modules.

Reply:
Yep! Will do. I think Rick just created a BlobTestUtils so I can move them there

Reply:
Done!